lib/apiservers/engine/backends/eventmonitor.go

```diff
@@ -243,11 +249,14 @@ func (p DockerEventPublisher) PublishEvent(event plevents.BaseEvent) {
 			attrs["exitCode"] = code
 			actor := CreateContainerEventActorWithAttributes(vc, attrs)
 			EventService().Log(containerDieEvent, eventtypes.ContainerEventType, actor)
+			// TODO: this really, really shouldn't be in the event publishing code - it's fine to have multiple consumers of events
+			// and this should be registered as a callback by the logic responsible for the MapPorts portion.
 			if err := UnmapPorts(vc.ContainerID, vc); err != nil {
 				log.Errorf("Event Monitor failed to unmap ports for container(%s): %s - eventID(%s)", containerShortID, err, event.ID)
 			}
 
 			// auto-remove if required
+			// TODO: this should be a separate event hook registered by logic outside of the publish events loop.
```

Comment:
Are these two tech debt? If so is there a new issue opened?

Reply:
Yup - the port layer tech debt is caught under the `vmomi gateway`/`consolidated cache and eventing` work but this is not.
#7176 and #7177 opened for this